salt/modules/saltutil.py

```diff
@@ -514,6 +514,30 @@ def sync_utils(saltenv=None, refresh=True):
     return ret
 
 
+def list_extmods():
+    '''
+    .. versionadded:: Nitrogen
+
+    List Salt modules which have been synced externally
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt '*' saltutil.list_extmods
+    '''
+    ret = {}
+    ext_dir = os.path.join(__opts__['cachedir'], 'extmods')
+    mod_types = os.listdir(ext_dir)
+    for mod_type in mod_types:
+        ret[mod_type] = set()
+        for _, _, files in os.walk(os.path.join(ext_dir, mod_type)):
+            for file in files:
```

Comment:
We might want to change `file` to something else here, since that is a reserved word. I'm kind of surprised surprised pylint didn't say anything about that.